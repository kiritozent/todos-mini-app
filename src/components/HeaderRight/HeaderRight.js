import React from 'react';
import PropTypes from 'prop-types';
import { Input } from 'antd';
import { SearchOutlined, UserOutlined } from '@ant-design/icons';

const HeaderRight = props => {
  const { userOnClick, searchText, fetching, setSearchText, isMobile } = props;

  return (
    <div className="headerRight">
      <Input
        value={searchText}
        placeholder="Search Todos..."
        onChange={e => setSearchText(e.target.value)}
        prefix={<SearchOutlined />}
        allowClear
        disabled={fetching}
      />
      <UserOutlined
        className="userIcon"
        style={{
          marginRight: !isMobile && 16,
        }}
        onClick={() => userOnClick()}
      />
    </div>
  );
};

HeaderRight.propTypes = {
  userOnClick: PropTypes.func,
  searchText: PropTypes.string,
  fetching: PropTypes.bool,
  setSearchText: PropTypes.func,
  isMobile: PropTypes.bool,
};

HeaderRight.defaultProps = {
  userOnClick: () => {},
  searchText: '',
  fetching: false,
  setSearchText: () => {},
  isMobile: false,
};

export default HeaderRight;
