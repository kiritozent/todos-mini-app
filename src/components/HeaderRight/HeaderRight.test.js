import React from 'react';
import { shallow } from 'enzyme';
import HeaderRight from './HeaderRight';

describe('HeaderRight', () => {
  it('render without crashing passed', () => {
    const testCase = shallow(<HeaderRight />);
    expect(testCase).toMatchSnapshot();
  });
});
