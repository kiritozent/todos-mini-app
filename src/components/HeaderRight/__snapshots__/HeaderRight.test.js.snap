// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRight render without crashing passed 1`] = `
<div
  className="headerRight"
>
  <Input
    allowClear={true}
    disabled={false}
    onChange={[Function]}
    placeholder="Search Todos..."
    prefix={<ForwardRef(SearchOutlined) />}
    type="text"
    value=""
  />
  <ForwardRef(UserOutlined)
    className="userIcon"
    onClick={[Function]}
    style={
      Object {
        "marginRight": 16,
      }
    }
  />
</div>
`;
